JSQMessagesViewController/Controllers/JSQMessagesViewController.m

```diff
@@ -916,23 +916,24 @@ - (void)jsq_didReceiveKeyboardWillChangeFrameNotification:(NSNotification *)noti
 {
     NSDictionary *userInfo = [notification userInfo];
 
+    CGRect keyboardBeginFrame = [[[notification userInfo] objectForKey:UIKeyboardFrameBeginUserInfoKey] CGRectValue];
     CGRect keyboardEndFrame = [userInfo[UIKeyboardFrameEndUserInfoKey] CGRectValue];
 
     if (CGRectIsNull(keyboardEndFrame)) {
         return;
     }
-
+    
     UIViewAnimationCurve animationCurve = [userInfo[UIKeyboardAnimationCurveUserInfoKey] integerValue];
     NSInteger animationCurveOption = (animationCurve << 16);
 
     double animationDuration = [userInfo[UIKeyboardAnimationDurationUserInfoKey] doubleValue];
-
+    float offset = keyboardBeginFrame.origin.y - keyboardEndFrame.origin.y + self.inputToolbar.frame.size.height;
```

Comment:
- use `CGFloat`
- instead of accessing origin and height, let's use `CGRectGetMinY()` and `CGRectGetHeight()`